src/Compilers/CSharp/Portable/Utilities/ValueSetFactory.EqualableValueTC.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal static partial class ValueSetFactory
+    {
+        /// <summary>
+        /// A type class for values that can be directly compared using equality.
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        private interface EqualableValueTC<T>
```

Comment:
>EqualableValueTC [](start = 26, length = 16)

This interface name is a little confusing, given the one method it contains. Would it make sense to have a more specific name such as HasConstantsTC?

Reply:
This typeclass is for values that can be compared directly using `object.Equals()`.  All of the typeclasses for the value sets share this method.  The name of this typeclass represents its abstract meaning, not the set of methods.  I'll try to clarify in the comments.

---
In reply to: [378578845](https://github.com/dotnet/roslyn/pull/41503#discussion_r378578845) [](ancestors = 378578845)